packages/web-components/fast-foundation/src/breadcrumb-item/breadcrumb-item.template.ts

```diff
@@ -0,0 +1,54 @@
+import { html, when, slotted } from "@microsoft/fast-element";
+import { BreadcrumbItem } from "./breadcrumb-item";
+import { Breadcrumb } from "../breadcrumb";
+import { endTemplate, startTemplate } from "../patterns/start-end";
+
+/**
+ * The template for the {@link @microsoft/fast-foundation#(BreadcrumbItem:class)} component.
+ * @public
+ */
+export const BreadcrumbItemTemplate = html<BreadcrumbItem>`
+    <div role="listitem" class="listitem" part="listitem">
+        ${startTemplate}
+        <slot
+            ${slotted({
+                property: "defaultSlottedNodes",
+                filter: value =>
+                    (value.nodeType === 3 && value.textContent!.trim().length !== 0) ||
+                    value.nodeType === 1 ||
+                    false,
+            })}
+        ></slot>
```

Comment:
Might be helpful to have some kind of detail as to why we need this. It seems related to the default slotted content below, but almost as though it's a bit of trickery. I noticed that when I removed this and just went down to a single slot, the control "broke" when passing start/end content. I assume this is why this is here. If that is the case, I think it would be helpful to add a bit of a note in the API docs or somewhere as to what is happening here. Without it, I think it'll cause confusion.

Reply:
Yeah, the `defaultSlottedNodes` slot, is to use as the default slot. The slot below used to be the default slot, but in order to get both the name attribute and SVG to render, I had to give it a slot name and create another slot used as a default.

Just to give some background context.
The structure I had before this was,
```ts
<slot>
            <a
                class="control"
                part="control"
                href="${x => x.href}"
                aria-current="${x => (x.isCurrent ? "page" : void 0)}"
            >
                ${startTemplate}
                <span class="content" part="content">
                    ${x => x.name}
                </span>
                ${endTemplate}
            </a>
</slot>
```
For some reason, every time I add a `start` or an `end slot, the `name` attribute disappears. I found that if the HTML is formatted this way,
```html
<fast-breadcrumb-item name="Home" href="#">
   <svg slot="start">
       <path ... />
   </svg>
</fast-breadcrumb-item>
```
Nothing gets rendered, but if I format the HTML this way,
```html
<fast-breadcrumb-item name="Home" href="#"><svg slot="start">
    <path ... />
</svg></fast-breadcrumb-item>
```
then I see the icon and the name attribute. Noticed that both the component tag and the SVG tag need to be on the same line.
I am still figuring out why putting both tags on the same line is working. One theory is that in the breadcrumb item template, the anchor is inside of the default <slot>. That means that any nodes, including spaces, replace the content.
Hoping if anyone has seen this before and why having the tags on the same line is working.

Reply:
added `defaultSlottedNodes ` to the spec

Reply:
I think we'll just want to document this in code as well. It's not apparent until you try to remove it why it's there.

Reply:
After more investigation and understanding why this is needed and how we can avoid having to do this, I have updated the template to use the `when` directive. If the breadcrumb-item has an href it will be anchor.